core/src/main/java/foundation/stack/datamill/configuration/Wiring.java

```diff
@@ -265,6 +283,24 @@ public Wiring addFormatted(String name, String format, Object... arguments) {
         }
     }
 
+    /**
+     * Check if the wiring already contains an instance of the provided class. If it exists, return it, otherwise, delegate
+     * to {@link #constructWith(Class, Class[])} in order to construct it.
+     *
+     * @param clazz          Class we want to create an instance of.
+     * @param parameterTypes Parameter types to be matched in constructor to be used.
+     * @return Instance that was constructed.
+     * @throws IllegalArgumentException If the class is an interface, abstract class or has no public constructors.
+     * @throws IllegalStateException    If all dependencies for constructing an instance cannot be satisfied.
+     */
+    public <T> T constructIfMissing(Class<T> clazz, Class<?>... parameterTypes) {
```

Comment:
We should call this constructWithIfMissing to match the pattern used with construct and constructWith